### PR TITLE
VIVO-1960: GitHub Build Action

### PR DESCRIPTION
**Pull request by wwelling**
Resolves [VIVO-1960](https://jira.lyrasis.org/browse/VIVO-1960)

https://github.com/vivo-project/Vitro/pull/210

# What does this pull request do?
Replaces Travis CI with GitHub Actions.

# What's new?
`.github/workflows/build.yml`

@VIVO-project/vivo-committers

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-cache-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-cache-m2-
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```
Comment (awoods) [inline .github/workflows/build.yml:28]: Is this "openjdk"... which is what we want.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Build
+
+on: [ push, pull_request, workflow_dispatch ]
```
Comment (awoods) [inline .github/workflows/build.yml:3]: What did you have in mind with `workflow_dispatch` : https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch ?

File: .github/workflows/build.yml (hunk 1 above)
Comment (wwelling) [reply .github/workflows/build.yml:28]: https://github.com/actions/setup-java

Zulu Community distribution of OpenJDK

File: .github/workflows/build.yml (hunk 2 above)
Comment (wwelling) [reply .github/workflows/build.yml:3]: After being merged into main, will have the ability to run workflow without push or PR. This is useful for advanced integration testing. Such as if Vitro default branch changes, you can trigger a build CI of VIVO without a push or PR.

**Review by awoods — APPROVED** [review]

**Pull request merged**